README.md

```diff
@@ -13,3 +13,81 @@ which is then inserted into the HoloPortOS instance.  When the device boots, it
 - Start the Holo services
 - Eject the USB and blacklist the kernel modules
 
+## Building & Geneating a `HoloPortConfiguration`
+
+We'll generate a `HoloPortConfiguration` object in JSON form, into `holo.json`:
+
+```
+$ nix-build -A holo-configure
+$ ./result/bin/holo-configure --name "HP1" --email "a@b.ca" --password "secret" | tee holo.json
+```
+
+If that doesn't work, try using the nix-shell and manual build approach:
+```
+$ nix-shell
+$ cargo build --release --lib --bin holo-configure
+
+$ ./target/release/holo-configure --name "HP1" --email "a@b.ca" --password "secret" | tee holo.json
+Generating HoloPort Configuration for email: a@b.ca
+{
+  "name": "HP1",
+  "email": "a@b.ca",
+  "admin_pubkey": "HcACj5GG78Nfa476fdvcAXwOQdv7hq7gyHi6bueg7ZSb4iix5hNpUcDFjnjejvi",
+  "seed_key": "HcBciRfTSA9E9h6y9mdfzi95PPYhkj6qfBHsXRAC34hpqjx3kxHNIWKYzexuzva",
+  "seed": "HcCcjf5ebsEDxmiz8j7YZX3r47qriFhwwPFZc7Vc7Wys5ozcFwUAQQTXoaygnbr",
+  "seed_sig": "WO0PYkFg1RZEP1UOzdBacj5QtHuM37uqjn0zPSSsgw8gJX2TU4NoQNb3tDMNvSFK5n4dDcen10ScGsRIde5iCA=="
+}
+```
+
+### `name`
+
+Optionally, make the `admin_key` and `seed_key` unique, by hashing the supplied `name` into
+`password` when performing the Argon2 password hashing.
+
+### `email` and `password`
+
+These will be harvested from stdin, if not supplied on the command-line.
+
+The `email` is used as a salt (SHA-2 256 hashed, to avoid too-short email addresses).
+
+The `password` is hashed (optionally, with any supplied `name`), and the resultant salt and password
+hashes are used to generate an argon2 password hash, which is used to generate the Admin and
+Blinding keypairs.
+
+### `admin_pubkey: String`: "HcAcj..."
+
+The Admin Private key is used to sign all HoloPort admin requests; the `admin_pubkey` is used to
+authenticate them.
+
+### `seed_key: Option<String>`: "HcBci..."
+
+A 256-bit AES ECB encryption key is *always* used to encrypt the seed; it is *optionally* supplied
+here in the config.
```

Comment:
I'm uncomfortable with using AES, and this 200% shouldn't be ECB. Please let's remove this for now, we don't have any use for that at this moment and designing this correctly will take time. I'm almost sure that this is not the right direction, and we don't have time for arguing about this right now. Let's agree on the most minimal thing.

Reply:
To elaborate, using `seed_key` the way you previously explained would require fundamental redesign of boot sequence, to the point NixOS/systemd won't be able to handle it, and layers of infrastructure that is not there in order to access HoloPort to unlock it.

There are other ways to solve this that don't require any of that: for example, we could make `holochain-conductor` handle Holochain key decryption. If you don't like that physical access to flash drive is too easy, we can tether the two by encrypting `holo.json` with key persisted on hard drive on first boot.

We never had this discussion, and I feel we should have it before we commit to doing this.

Reply:
Ya, I'm using ECB just because it was the shortest path to just using the AES block cipher exactly once (the lower-level block cipher API wasn't easily available).  Are you opposed to AES because of its design heritage (US security researchers), or because it has possible flaws?  Any 256-bit block cipher will do, to avoid changing the seed's plaintext vs. ciphertext size.

However, shipping the seed material in the clear is very dangerous, as it is an obvious security lapse in judgement; at least by encrypting it and optionally (always, for now) providing the key, we are clearly defining that we didn't just accidentally screw up our design...

Reply:
> Ya, I'm using ECB just because it was the shortest path to just using the AES block cipher exactly once (the lower-level block cipher API wasn't easily available). Are you opposed to AES because of its design heritage (US security researchers), or because it has possible flaws? Any 256-bit block cipher will do, to avoid changing the seed's plaintext vs. ciphertext size.

In this case where `len(key)` is equal to `len(msg)`, I think the best solution would be to go for information-theoretically secure XOR one-time pad, which would be an improvement in this case because none of AES cryptanalysis will apply.

I actually misjudged this initially, because it's very rare that using AES-ECB is not completely broken, sorry for that. This is one of such cases, but in this case, one-time pad seems to be an improvement.

> However, shipping the seed material in the clear is very dangerous, as it is an obvious security lapse in judgement; at least by encrypting it and optionally (always, for now) providing the key, we are clearly defining that we didn't just accidentally screw up our design...

Right now, it's not an improvement because we ship key alongside the seed. I agree it leaves more to be desired, but design with persisting keys on hard drive was also vulnerable to physical attacks. I believe that, as we move forward, it will become more obvious which seed protection scheme we should go for.